src/averageAges.js

```diff
@@ -38,6 +52,26 @@ function calculateMenAverageAge(people, century) {
  */
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+  const filteredWomen = people.filter((person) => person.sex === 'f');
+  const filteredMothers = [];
+
+  for (const key of filteredWomen) {
+    const checkCondition = people.some(women => key.name === women.mother);
+
+    if (checkCondition) {
+      filteredMothers.push(key);
+    }
+  };
+
+  function checkWhoToCompare(compare) {
+    return compare
+      .map(age => age.died - age.born)
+      .reduce((a, b) => a + b, 0) / compare.length;
+  }
```

Comment:
1. Adjust names to be more meaningful
2. What about moving this function out of `calculateWomenAverageAge` and reuse it in `calculateMenAverageAge` as well?
```suggestion
  function getAverageAge(people) {
    return people
      .map(person=> person.died - person.born)
      .reduce((prevAge, nextAge) => prevAge + nextAge, 0) / people.length;
  }
```